Assets/Scripts/PlayerManager.cs

```diff
@@ -47,28 +48,46 @@ private void Start()
 
     private void Update()
     {
-        if (CanRest)
+        if (CanRest && Input.GetKeyDown(KeyCode.F))
         {
-            bool isResting = Input.GetKey(KeyCode.E);
-            UIManager.Instance.UpdateRestingText(isResting);
+            // Temporary Controls for Minigame
+            MinigameManager.Instance.LoadRandomMinigame();   
+        }
+    }
 
-            if (isResting)
-            {
-                HandleIncreaseStamina();
-            } 
-            else
-            {
-                // Temporary Controls for Minigame
-                if (Input.GetKeyDown(KeyCode.P))
-                {
-                    HandleTriggerStartMinigame();
-                }
-            }
+    private void FixedUpdate()
+    {
+        HandleDecreaseStamina();
+
+        float minDistance = distToClosestEnemy();
+
+        if (minDistance < dangerRadius)
+        {
+            HandleIncreaseStamina((dangerRadius - minDistance) / dangerRadius);
         }
-        else
+
+        // Temporary Controls for Minigame
+        if (Input.GetKeyDown(KeyCode.P))
+        {
+            HandleTriggerStartMinigame();
+        }
+    }
+
+    private float distToClosestEnemy()
```

Comment:
maybe rename to like `DistanceToClosestEnemy` (capitalized first letter being my main issue here though so up to you on the name)